### PR TITLE
wifi: Basic AP mode support

### PR DESCRIPTION
**Pull request by krish2718**
Beaconing only

### CODE REVIEW

File: modules/hostap/src/supp_api.c [hunk 1]
```diff
@@ -398,14 +380,47 @@ int z_wpa_supplicant_connect(const struct device *dev,
...
+			   "Interface %s is down, dropping connect",
+			   dev->name);
+		return -1;
+	}
+
+	k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+
+	wpa_s = get_wpa_s_handle(dev);
+	if (!wpa_s) {
+		ret = -1;
```
Comment (rado17) [inline modules/hostap/src/supp_api.c:452]: Can we set `ret` to `-ENODEV` instead of `-1`?

File: drivers/wifi/nrf700x/src/net_if.c [hunk 2]
```diff
@@ -38,6 +38,35 @@ static struct net_mgmt_event_callback ip_maddr4_cb;
...
+		goto out;
+	}
+
+	if (!event) {
+		LOG_ERR("%s: event is NULL",
+			__func__);
+		goto out;
+	}
+
+	(void)event_len;
```
Comment (rado17) [inline drivers/wifi/nrf700x/src/net_if.c:59]: What does this do?

File: modules/hostap/src/supp_api.c (hunk 1 above)
Comment (krish2718) [reply modules/hostap/src/supp_api.c:452]: Sure, I guess we need to fix this at all places, is it okay if I do the cleanup together in another PR?

File: drivers/wifi/nrf700x/src/net_if.c (hunk 2 above)
Comment (krish2718) [reply drivers/wifi/nrf700x/src/net_if.c:59]: Tells the compiler that `not using this variable is intended and not a bug` and then we don't see `unused variable` warning.

File: modules/hostap/src/supp_api.c (hunk 1 above)
Comment (rado17) [reply modules/hostap/src/supp_api.c:452]: Sure.

**Review by rado17 — APPROVED** [review]

**Pull request merged**